include/envoy/tracing/http_tracer.h

```diff
@@ -7,42 +7,74 @@
 namespace Tracing {
 
 /*
- * Context used by tracers, it carries additional data needed to populate the trace.
+ * Tracing configuration, it carries additional data needed to populate the span.
+ */
+class TracingConfig {
+public:
+  virtual ~TracingConfig() {}
+
+  virtual const std::string& operationName() const PURE;
+  virtual const std::string& serviceNode() const PURE;
+};
+
+/*
+ * Basic abstraction for span.
+ */
+class Span {
+public:
+  virtual ~Span() {}
+
+  virtual void setTag(const std::string& name, const std::string& value) PURE;
+  virtual void finishSpan() PURE;
+};
+
+typedef std::shared_ptr<Span> SpanPtr;
+
+/**
+ * Tracing context, it carries currently active span.
  */
 class TracingContext {
```

Comment:
from further looking, the existence of this class at all I think is strange. I would remove it. I would just have the active stream hold a SpanPtr (or not).